Groups: Crash on nil pointer deref

### What happened?

The provider crashes on nil pointer deref with the following logs:

```log
1.6880365645680258e+09  DEBUG   provider-gitlab Starting        {"sync-period": "1h0m0s"}
I0629 11:02:45.618701       1 request.go:682] Waited for 1.032956198s due to client-side throttling, not priority and fairness, request: GET:https://172.20.0.1:443/apis/acm.aws.crossplane.io/v1alpha1?timeout=32s
1.6880365708763983e+09  INFO    controller-runtime.metrics      Metrics server is starting to listen    {"addr": ":8080"}
1.688036570878509e+09   INFO    Starting server {"path": "/metrics", "kind": "metrics", "addr": "[::]:8080"}
1.688036570878656e+09   INFO    controller.providerconfig/providerconfig.gitlab.crossplane.io   Starting EventSource    {"reconciler group": "gitlab.crossplane.io", "reconciler kind": "ProviderConfig", "source": "kind source: *v1beta1.ProviderConfig"}
1.6880365708786852e+09  INFO    controller.providerconfig/providerconfig.gitlab.crossplane.io   Starting EventSource    {"reconciler group": "gitlab.crossplane.io", "reconciler kind": "ProviderConfig", "source": "kind source: *v1beta1.ProviderConfigUsage"}
1.688036570878692e+09   INFO    controller.providerconfig/providerconfig.gitlab.crossplane.io   Starting Controller     {"reconciler group": "gitlab.crossplane.io", "reconciler kind": "ProviderConfig"}
1.688036570879072e+09   INFO    controller.managed/member       Starting EventSource    {"reconciler group": "groups.gitlab.crossplane.io", "reconciler kind": "Member", "source": "kind source: *v1alpha1.Member"}
1.6880365708791082e+09  INFO    controller.managed/member       Starting Controller     {"reconciler group": "groups.gitlab.crossplane.io", "reconciler kind": "Member"}
1.6880365708791935e+09  INFO    controller.managed/group        Starting EventSource    {"reconciler group": "groups.gitlab.crossplane.io", "reconciler kind": "Group", "source": "kind source: *v1alpha1.Group"}
1.6880365708792179e+09  INFO    controller.managed/group        Starting Controller     {"reconciler group": "groups.gitlab.crossplane.io", "reconciler kind": "Group"}
1.688036570879277e+09   INFO    controller.managed/project      Starting EventSource    {"reconciler group": "projects.gitlab.crossplane.io", "reconciler kind": "Project", "source": "kind source: *v1alpha1.Project"}
1.6880365708793004e+09  INFO    controller.managed/project      Starting Controller     {"reconciler group": "projects.gitlab.crossplane.io", "reconciler kind": "Project"}
1.6880365708793702e+09  INFO    controller.managed/deploytoken  Starting EventSource    {"reconciler group": "groups.gitlab.crossplane.io", "reconciler kind": "DeployToken", "source": "kind source: *v1alpha1.DeployToken"}
1.6880365708793898e+09  INFO    controller.managed/deploytoken  Starting Controller     {"reconciler group": "groups.gitlab.crossplane.io", "reconciler kind": "DeployToken"}
1.6880365708794286e+09  INFO    controller.managed/member       Starting EventSource    {"reconciler group": "projects.gitlab.crossplane.io", "reconciler kind": "Member", "source": "kind source: *v1alpha1.Member"}
1.68803657087945e+09    INFO    controller.managed/member       Starting Controller     {"reconciler group": "projects.gitlab.crossplane.io", "reconciler kind": "Member"}
1.6880365708797495e+09  INFO    controller.managed/hook Starting EventSource    {"reconciler group": "projects.gitlab.crossplane.io", "reconciler kind": "Hook", "source": "kind source: *v1alpha1.Hook"}
1.6880365708798149e+09  INFO    controller.managed/hook Starting Controller     {"reconciler group": "projects.gitlab.crossplane.io", "reconciler kind": "Hook"}
1.6880365708799133e+09  INFO    controller.managed/variable     Starting EventSource    {"reconciler group": "projects.gitlab.crossplane.io", "reconciler kind": "Variable", "source": "kind source: *v1alpha1.Variable"}
1.6880365708799403e+09  INFO    controller.managed/variable     Starting Controller     {"reconciler group": "projects.gitlab.crossplane.io", "reconciler kind": "Variable"}
1.6880365708802931e+09  INFO    controller.managed/deploytoken  Starting EventSource    {"reconciler group": "projects.gitlab.crossplane.io", "reconciler kind": "DeployToken", "source": "kind source: *v1alpha1.DeployToken"}
1.6880365708803213e+09  INFO    controller.managed/deploytoken  Starting Controller     {"reconciler group": "projects.gitlab.crossplane.io", "reconciler kind": "DeployToken"}
1.688036570979731e+09   INFO    controller.managed/project      Starting workers        {"reconciler group": "projects.gitlab.crossplane.io", "reconciler kind": "Project", "worker count": 1}
1.6880365709801288e+09  DEBUG   provider-gitlab Reconciling     {"controller": "managed/project", "request": "/<redacted>"}
1.6880365709876876e+09  INFO    controller.managed/deploytoken  Starting workers        {"reconciler group": "projects.gitlab.crossplane.io", "reconciler kind": "DeployToken", "worker count": 1}
1.688036570991415e+09   DEBUG   provider-gitlab Reconciling     {"controller": "managed/deploytoken", "request": "/<redacted>"}
1.6880365709878244e+09  INFO    controller.managed/variable     Starting workers        {"reconciler group": "projects.gitlab.crossplane.io", "reconciler kind": "Variable", "worker count": 1}
1.6880365709878945e+09  INFO    controller.managed/group        Starting workers        {"reconciler group": "groups.gitlab.crossplane.io", "reconciler kind": "Group", "worker count": 1}
1.6880365709964132e+09  DEBUG   provider-gitlab Reconciling     {"controller": "managed/group", "request": "/openhoa-members"}
1.688036570987856e+09   INFO    controller.managed/hook Starting workers        {"reconciler group": "projects.gitlab.crossplane.io", "reconciler kind": "Hook", "worker count": 1}
1.688036570988033e+09   INFO    controller.managed/deploytoken  Starting workers        {"reconciler group": "groups.gitlab.crossplane.io", "reconciler kind": "DeployToken", "worker count": 1}
1.6880365709798114e+09  INFO    controller.providerconfig/providerconfig.gitlab.crossplane.io   Starting workers        {"reconciler group": "gitlab.crossplane.io", "reconciler kind": "ProviderConfig", "worker count": 1}
1.6880365709993916e+09  DEBUG   provider-gitlab Reconciling     {"controller": "providerconfig/providerconfig.gitlab.crossplane.io", "request": "/<redacted>"}
1.688036571049716e+09   DEBUG   provider-gitlab Reconciling     {"controller": "providerconfig/providerconfig.gitlab.crossplane.io", "request": "/default"}
1.6880365711964777e+09  INFO    controller.managed/member       Starting workers        {"reconciler group": "groups.gitlab.crossplane.io", "reconciler kind": "Member", "worker count": 1}
1.6880365712002149e+09  DEBUG   provider-gitlab Reconciling     {"controller": "managed/member", "request": "/atots-atots-members-7243"}
1.6880365712067313e+09  INFO    controller.managed/member       Starting workers        {"reconciler group": "projects.gitlab.crossplane.io", "reconciler kind": "Member", "worker count": 1}
1.6880365713188407e+09  DEBUG   provider-gitlab External resource is up to date {"controller": "managed/member", "request": "/atots-atots-members-7243", "uid": "fb0da961-d73c-4696-998e-89b6b4bb6f98", "version": "938692341", "external-name": "127378", "requeue-after": 1688036631.3188386}
1.6880365713266778e+09  DEBUG   provider-gitlab Reconciling     {"controller": "managed/member", "request": "/bbpneo-bbpneo-members-3030"}
1.6880365713489234e+09  DEBUG   provider-gitlab External resource is up to date {"controller": "managed/deploytoken", "request": "/<redacted>", "uid": "e285cde2-3f47-4422-b28f-9b958ef67b48", "version": "868785028", "external-name": "1224", "requeue-after": 1688036631.3489203}
1.6880365713541274e+09  DEBUG   provider-gitlab External resource is up to date {"controller": "managed/group", "request": "/openhoa-members", "uid": "e2accc62-2b42-4db3-b557-a6d6b0f54a80", "version": "938689383", "external-name": "127376", "requeue-after": 1688036631.3541245}
1.6880365713551168e+09  DEBUG   provider-gitlab Reconciling     {"controller": "managed/deploytoken", "request": "/<redacted>"}
1.6880365713605552e+09  DEBUG   provider-gitlab Reconciling     {"controller": "managed/group", "request": "/atots"}
1.6880365715434258e+09  DEBUG   provider-gitlab External resource is up to date {"controller": "managed/member", "request": "/bbpneo-bbpneo-members-3030", "uid": "6ef8af7b-6031-4602-9ff1-8a1f82d49d31", "version": "938692378", "external-name": "127377", "requeue-after": 1688036631.543422}
1.6880365715515068e+09  DEBUG   provider-gitlab Reconciling     {"controller": "managed/member", "request": "/openhoa-openhoa-members-17376"}
1.6880365715629826e+09  DEBUG   provider-gitlab External resource is up to date {"controller": "managed/deploytoken", "request": "/<redacted>", "uid": "b778f786-0691-4f32-9fa0-042f57c5d4ae", "version": "868785056", "external-name": "1246", "requeue-after": 1688036631.5629797}
1.6880365715715885e+09  DEBUG   provider-gitlab Reconciling     {"controller": "managed/deploytoken", "request": "/<redacted>"}
1.6880365717086618e+09  DEBUG   provider-gitlab External resource is up to date {"controller": "managed/deploytoken", "request": "/<redacted>", "uid": "1d484ed5-0768-49c9-80e4-69a17cda1643", "version": "868784767", "external-name": "1256", "requeue-after": 1688036631.708658}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x145577a]
goroutine 770 [running]:
github.com/crossplane-contrib/provider-gitlab/pkg/clients/groups.GenerateObservation(0xc001b35a40)
        github.com/crossplane-contrib/provider-gitlab/pkg/clients/groups/group.go:118 +0x5ba
github.com/crossplane-contrib/provider-gitlab/pkg/controller/groups.(*external).Observe(0xc001bdb8a0, {0x1a4e688?, 0xc001c21b00?}, {0x1a5b788?, 0xc001d8c2c0})
        github.com/crossplane-contrib/provider-gitlab/pkg/controller/groups/group.go:113 +0x245
github.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*Reconciler).Reconcile(0xc0001080b0, {0x1a4e6c0, 0xc001c45a10}, {{{0x0?, 0x172ddc0?}, {0xc0007b8b10?, 0x30?}}})
        github.com/crossplane/crossplane-runtime@v0.18.0/pkg/reconciler/managed/reconciler.go:767 +0x229f
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0xc000108160, {0x1a4e6c0, 0xc001c45980}, {{{0x0?, 0x172ddc0?}, {0xc0007b8b10?, 0x4041f4?}}})
        sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:114 +0x27e
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000108160, {0x1a4e618, 0xc00048f640}, {0x1656b40?, 0xc00092ba00?})
        sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:311 +0x349
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000108160, {0x1a4e618, 0xc00048f640})
        sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
        sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
        sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:223 +0x31c
```

The error appears to be in this line: https://github.com/crossplane-contrib/provider-gitlab/blob/master/pkg/clients/groups/group.go#L118

### How can we reproduce it?

n.a.

### What environment did it happen in?

provider-gitlab v0.2.0

--- a/pkg/clients/groups/group.go
+++ b/pkg/clients/groups/group.go
@@ -115,7 +115,10 @@ func GenerateObservation(grp *gitlab.Group) v1alpha1.GroupObservation { // nolin
 			group.SharedWithGroups[i].GroupName = c.GroupName
 			group.SharedWithGroups[i].GroupFullPath = c.GroupFullPath
 			group.SharedWithGroups[i].GroupAccessLevel = c.GroupAccessLevel
-			group.SharedWithGroups[i].ExpiresAt = &metav1.Time{Time: time.Time(*c.ExpiresAt)}
+
+			if c.ExpiresAt != nil {
+				group.SharedWithGroups[i].ExpiresAt = &metav1.Time{Time: time.Time(*c.ExpiresAt)}
+			}
 		}
 	}
 

--- a/pkg/clients/groups/group_test.go
+++ b/pkg/clients/groups/group_test.go
@@ -171,6 +171,30 @@ func TestGenerateObservation(t *testing.T) {
 				CreatedAt:           v1alpha1CreatedAt,
 			},
 		},
+		"SharedWithGroupExpiresAtIsNil": {
+			args: args{
+				p: &gitlab.Group{
+					SharedWithGroups: []struct {
+						GroupID          int             `json:"group_id"`
+						GroupName        string          `json:"group_name"`
+						GroupFullPath    string          `json:"group_full_path"`
+						GroupAccessLevel int             `json:"group_access_level"`
+						ExpiresAt        *gitlab.ISOTime `json:"expires_at"`
+					}{
+						{
+							ExpiresAt: nil,
+						},
+					},
+				},
+			},
+			want: v1alpha1.GroupObservation{
+				SharedWithGroups: []v1alpha1.SharedWithGroups{
+					{
+						ExpiresAt: nil,
+					},
+				},
+			},
+		},
 	}
 
 	for name, tc := range cases {

